cartography/intel/aws/ecs.py

```diff
@@ -0,0 +1,590 @@
+            c.health_status = container.healthStatus,
+            c.cpu = container.cpu,
+            c.memory = container.memory,
+            c.memory_reservation = container.memoryReservation,
+            c.gpu_ids = container.gpuIds,
+            c.lastupdated = {aws_update_tag}
+        WITH c
+        MATCH (t:ECSTask{id: c.task_arn})
+        MERGE (t)-[r:HAS_CONTAINER]->(c)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+    """
+    neo4j_session.run(
+        ingest_containers,
+        Containers=data,
+        Region=region,
+        AWS_ACCOUNT_ID=current_aws_account_id,
+        aws_update_tag=aws_update_tag,
+    )
+
+
+@timeit
+def cleanup_ecs(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
+    run_cleanup_job('aws_import_ecs_cleanup.json', neo4j_session, common_job_parameters)
+
+
+@timeit
+def sync(
+    neo4j_session: neo4j.Session, boto3_session: boto3.session.Session, regions: List[str], current_aws_account_id: str,
+    update_tag: int, common_job_parameters: Dict,
+) -> None:
+    for region in regions:
+        logger.info("Syncing ECS for region '%s' in account '%s'.", region, current_aws_account_id)
+        cluster_arns, clusters = get_ecs_clusters(boto3_session, region)
+        if len(clusters) == 0:
+            continue
+        load_ecs_clusters(neo4j_session, clusters, region, current_aws_account_id, update_tag)
+        for cluster_arn in cluster_arns:
+            cluster_instances = get_ecs_container_instances(
+                cluster_arn,
+                boto3_session,
+                region,
+            )
+            load_ecs_container_instances(
+                neo4j_session,
+                cluster_arn,
+                cluster_instances,
+                region,
+                current_aws_account_id,
+                update_tag,
+            )
+            task_definitions = get_ecs_task_definitions(
+                boto3_session,
+                region,
+            )
+            load_ecs_task_definitions(
+                neo4j_session,
+                task_definitions,
+                region,
+                current_aws_account_id,
+                update_tag,
+            )
+            services = get_ecs_services(
+                cluster_arn,
+                boto3_session,
+                region,
+            )
+            load_ecs_services(
+                neo4j_session,
+                cluster_arn,
+                services,
+                region,
+                current_aws_account_id,
+                update_tag,
+            )
+            tasks = get_ecs_tasks(
+                cluster_arn,
+                boto3_session,
+                region,
+            )
+            load_ecs_tasks(
+                neo4j_session,
+                cluster_arn,
+                tasks,
+                region,
+                current_aws_account_id,
+                update_tag,
+            )
```

Comment:
Not a blocker but just a suggestion: How about we add in a new function for these lines of code so that the sync() function remains short and only has the regions loop? Would just help improve the readability imo.

Reply:
I think if we were unit testing modules this would be a good step for testibility, but I actually think from a reading perspective it's easier to see the actions in a row. I'm not opposed to it, but I'd rather do this in a refactor later :)

Reply:
Sure no problem!